setup.py

```diff
@@ -148,15 +148,17 @@
     install_requires=['numpy',
                       'matplotlib',
                       'scipy',
-                      'Sphinx',
                       'numpydoc',
                       'dill',
                       'joblib',
-                      'future',  # Optional
                       'funcsigs',
-                      'jupyter',
-                      'nbsphinx',
-                      'recommonmark'
+                      'jupyter'
+    ],
+    extra_requires=[
```

Comment:
this should be a dictionary.
`extra_requires = {'dev': [list of dependencies]}

Reply:
thanks for the review. corrections in latest commit.